src/pages/Browse/index.tsx

```diff
@@ -21,39 +32,61 @@ type PropsFromRouter = {
 
 type PropsFromState = {
   apiState: ApiState;
+  file: VersionFile | null | void;
   version: Version;
 };
 
 /* eslint-disable @typescript-eslint/indent */
-type Props = RouteComponentProps<PropsFromRouter> &
+export type Props = RouteComponentProps<PropsFromRouter> &
   PropsFromState &
+  PublicProps &
   ConnectedReduxProps;
 /* eslint-enable @typescript-eslint/indent */
 
 export class BrowseBase extends React.Component<Props> {
-  async componentDidMount() {
-    const { apiState, dispatch, match } = this.props;
-    const { addonId, versionId } = match.params;
+  static defaultProps = {
+    _fetchVersion: fetchVersion,
+    _log: log,
+  };
 
-    const response = (await getVersion({
-      addonId: parseInt(addonId, 10),
-      apiState,
-      versionId: parseInt(versionId, 10),
-    })) as ExternalVersion;
+  componentDidMount() {
+    const { _fetchVersion, dispatch, match } = this.props;
+    const { addonId, versionId } = match.params;
 
-    if (response && response.id) {
-      dispatch(versionActions.loadVersionInfo({ version: response }));
-    }
+    dispatch(
+      _fetchVersion({
+        addonId: parseInt(addonId, 10),
+        versionId: parseInt(versionId, 10),
+      }),
+    );
   }
 
   render() {
-    const { version } = this.props;
+    const { file, version } = this.props;
+
+    if (!version) {
+      return (
+        <Col>
+          <FontAwesomeIcon icon="spinner" spin />{' '}
+          {gettext('Loading version...')}
+        </Col>
+      );
+    }
 
     return (
       <React.Fragment>
-        <Col md="3">{version && <FileTree version={version} />}</Col>
-        <Col>
-          <p>Version ID: {this.props.match.params.versionId}</p>
+        <Col md="3">
+          <FileTree version={version} />
+        </Col>
+        <Col md="9">
+          {file ? (
+            <Highlight className="auto">{file.content}</Highlight>
```

Comment:
> This assumes a text file, but this could be an image. I was looking at #109, and I guess that will need to build on what you're doing here.

Yep, there is no support for that at all.

> Do you think it's worth extracting this out into a component at this point, or would you prefer to make that a separate issue as well?

I believe we can do that later when we know what we want to do. This patch only enables new features/issues.